src/ashley/receivers.py

```diff
@@ -0,0 +1,85 @@
+"""This module defines signal receivers for the Ashley application"""
+
+import logging
+
+from django.conf import settings
+from django.dispatch import receiver
+from django.utils.translation import to_locale
+from machina.apps.forum_conversation.signals import topic_viewed
+from machina.core.db.models import get_model
+from tincan import (
+    Activity,
+    ActivityDefinition,
+    Context,
+    ContextActivities,
+    LanguageMap,
+    Verb,
+)
+
+from .xapi import build_statement, get_course_extensions
+
+LTIContext = get_model("ashley", "LTIContext")
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(topic_viewed)
+# pylint: disable=unused-argument
+def track_topic_view(sender, topic, user, request, response, **kwargs):
+    """ Log a XAPI statement when a user views a topic. """
+
+    consumer = getattr(user, "lti_consumer", None)
+    if consumer is None:
+        logger.warning("Unable to get LTI consumer of user %s", user)
+        return
+    xapi_logger = logging.getLogger(f"xapi.{user.lti_consumer.slug}")
+
+    verb = Verb(
+        id="http://id.tincanapi.com/verb/viewed",
+        display=LanguageMap({"en-US": "viewed"}),
+    )
+
+    activity_extensions = {}
+    if request.forum_permission_handler.current_lti_context_id is not None:
+        try:
+            lti_context = LTIContext.objects.get(
+                pk=request.forum_permission_handler.current_lti_context_id
+            )
+            activity_extensions = get_course_extensions(lti_context)
```

Comment:
To avoid confusion, I would name it `activity_definition_extensions` or `object_extensions`.
```suggestion
            activity_definition_extensions = get_course_extensions(lti_context)
```
OR
```suggestion
            object_extensions = get_course_extensions(lti_context)
```

Reply:
Thanks! I have opted for `activity_definition_extensions`